### PR TITLE
chore: Add a new check ensure .editorconfig rules are honored

### PR DESCRIPTION
**Pull request by leomayleomay**
## Problem
This is to add a new pre-commit-hook check to ensure rules in .editorconfig will be honored, otherwise CI will fail

## Solution
Introduce a new git-hooks.nix as part of flake modules, it will host the `checks` which will run a pre-commit-hook to ensure rules in .editorconfig will be honored, otherwise CI will fail

### CODE REVIEW

**Comment by srid** [comment]
Thanks; is this ready for review? @leomayleomay 

Team - can we approve CI for this PR?

File: git-hooks.nix [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{ inputs, ... }:
+{
+  perSystem = { system, ... }:
+    {
+      checks.pre-commit-check = inputs.pre-commit-hooks.lib.${system}.run {
```
Comment (srid) [inline git-hooks.nix:5]: You don't have to manually create this check, because upstream repo provides its own flake-parts module that we can import.

Comment (leomayleomay) [reply git-hooks.nix:5]: @srid you mean [this](https://flake.parts/options/pre-commit-hooks-nix)?

**Comment by leomayleomay** [comment]
> Thanks; is this ready for review? @leomayleomay
> 
> Team - can we approve CI for this PR?

yes, this is ready for review, if the team could approve CI, I will see a CI failure due mal-formatted code

File: git-hooks.nix (hunk 1 above)
Comment (srid) [reply git-hooks.nix]: Yes.

**Comment by Datron** [comment]
@leomayleomay Thanks for the contribution! @srid can you approve too? I can merge after that.

**Comment by leomayleomay** [comment]
@Datron it's my pleasure :)

@srid will this check apply to .sql only or all the files?

**Comment by Datron** [comment]
@leomayleomay we follow [conventional commit guidelines](https://www.conventionalcommits.org/en/v1.0.0/) for our commits, can you add that? I think chore or fix is appropriate here.

**Comment by leomayleomay** [comment]
> @leomayleomay we follow [conventional commit guidelines](https://www.conventionalcommits.org/en/v1.0.0/) for our commits, can you add that? I think chore or fix is appropriate here.

@Datron thank you for letting me know :)

**Comment by leomayleomay** [comment]
@srid @Datron I've made changes to switch to [eclint](https://pkg.go.dev/gitlab.com/greut/eclint) and lint .sql files only, please let me know if there are other types of files to lint, cheers

**Comment by leomayleomay** [comment]
@srid the CI [failure](https://github.com/juspay/superposition/actions/runs/9223258312/job/25377253227#step:5:4208) seems to relate to devour-flake, please help out, I am running out of idea why it doesn't work for macos

File: .github/workflows/ci_check_pr.yaml
Comment (srid) [inline .github/workflows/ci_check_pr.yaml:1]: This file is redundant. See https://github.com/juspay/superposition/blob/main/.github/workflows/nix.yaml

Also, see https://github.com/srid/nixci (it already builds the checks)

**Comment by srid** [comment]
> @srid the CI [failure](https://github.com/juspay/superposition/actions/runs/9223258312/job/25377253227#step:5:4208) seems to relate to devour-flake, please help out, I am running out of idea why it doesn't work for macos

The issue is not with devour-flake but with the pre-commit-run run derivation which is failing to build. You can use `nix --accept-flake-config run github:srid/nixci build` in order to run CI locally.

**Comment by srid** [comment]
The pre-commit hook should be added to devShell shellHook too? Just like in `nammayatri`.

**Comment by leomayleomay** [comment]
> The pre-commit hook should be added to devShell shellHook too? Just like in `nammayatri`.

indeed, I thought this PR was all about CI, now added to devShell

File: .github/workflows/ci_check_pr.yaml
Comment (leomayleomay) [reply .github/workflows/ci_check_pr.yaml:1]: nice, this is removed now

**Comment by leomayleomay** [comment]
the pre-commit now works in devShell

image

File: .github/workflows/ci_check_pr.yaml [hunk 2]
```diff
@@ -1,131 +0,0 @@
-name: CI Checks on PRs
```
Comment (Datron) [inline .github/workflows/ci_check_pr.yaml:1]: @leomayleomay @srid we can't get rid of this file, we run E2E tests with newman.

Comment (leomayleomay) [reply .github/workflows/ci_check_pr.yaml:1]: @Datron it's my bad, I thought Srid meant this file redundant, I believe he meant my changes to this file redundant

**Review by Datron — APPROVED** [review]

**Comment by Datron** [comment]
@juspay/sdk-backend can you review?

File: .editorconfig [hunk 3]
```diff
@@ -12,6 +12,9 @@ trim_trailing_whitespace = true
 max_line_length = off
 trim_trailing_whitespace = false
 
+[*.sql]
+insert_final_newline = true
```
Comment (knutties) [inline .editorconfig:16]: We will need this for all files I guess.  Can make line 8 itself as true?

File: nix/pre-commit.nix [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{ ... }:
+{
+  perSystem = { pkgs, lib, ... }: {
+    pre-commit = {
+      check.enable = true;
+      settings = {
+        hooks = {
+            eclint.files = "\.sql$";
```
Comment (srid) [inline nix/pre-commit.nix:8]: In line with @knutties 's comment above, is there still a reason to limit eclint (EditoConfig linter) to sql files?

**Review by srid — COMMENTED** [review]
Approved from me, once we address @knutties 's comment above.

File: .editorconfig (hunk 3 above)
Comment (leomayleomay) [reply .editorconfig:16]: sure thing we could apply the rule to all files, let me see what I could do

**Review by Datron — APPROVED** [review]

**Comment by Datron** [comment]
@srid @knutties can you take a look?

**Review by ShubhranshuSanjeev — APPROVED** [review]

**Review by knutties — APPROVED** [review]

**Review by srid — APPROVED** [review]
👍

**Comment by leomayleomay** [comment]
@Datron please help to approve awaiting workflows, cheers

**Pull request merged**